types/v56/nsxt_types.go

```diff
@@ -0,0 +1,107 @@
+package types
+
+// OpenAPIEdgeGateway structure can hold both NSX-V and NSX-T edge gateways as returned by OpenAPI endpoint.
+//
+// Note. OpenAPI endpoint only allows to read NSX-V edge gateways.
+type OpenAPIEdgeGateway struct {
```

Comment:
```suggestion
type NxstEdgeGateway struct {
```

Reply:
This is a bigger discussion which got me. I am open for better names, but giving context below

The point of this PR is that we support NSX-T edge gateway (and I have created CRUD functions name *Nsxt*), but this structure can accommodate both types of edge gateways for READ (can create NSX-T only). Notable big difference is that `EdgeGatewayUplinks []EdgeGatewayUplinks` can be _only one_ in NSX-T case (and many in NSX-V case). 

Worth knowing - knowledge that there can only be one `[]EdgeGatewayUplinks` in NSX-T allowed me to remove one level of nesting in `vcd_nsxt_edgegateway` resource compared to `vcd_edgegateway` in Terraform which makes the resource easier to consume (and code).

So while We use this structure for NSX-T only - one could "read" NSX-V edge gateway.

cc @lvirbalas , @dataclouder

Reply:
I would vote explicitly make two types for NSXT and NSXV - then we would explicitly see differences even same API call would serialize to them.

Reply:
I think we should keep the name that Dainius has chosen